pkg/controller/controller_test.go

```diff
@@ -2193,3 +1770,218 @@ func getRecordedEvents(source <-chan string) []string {
 	}
 	return events
 }
+
+func assertNumEvents(t *testing.T, strings []string, number int) {
+	if e, a := number, len(strings); e != a {
+		t.Fatalf("Unexpected number of events: expected %v, got %v", e, a)
```

Comment:
it'd be helpful for this and the below functions to return an `error` so that failures show up on the right line